src/ddsrt/tests/dl.c

```diff
@@ -0,0 +1,224 @@
+  ddsrt_lib l;
+
+  r = ddsrt_dlopen("UnknownLib", false, &l);
+
+  CU_ASSERT_PTR_NULL_FATAL(l);
+
+  r = ddsrt_dlerror(buffer, sizeof(buffer));
+  CU_ASSERT_EQUAL_FATAL(r, DDS_RETCODE_OK);
+  printf("\n%s", buffer);
+}
+
+CU_Test(ddsrt_library, dlopen_null)
+{
+  char buffer[256];
+  dds_retcode_t r;
+  ddsrt_lib l;
+
+  r = ddsrt_dlopen(NULL, false, &l);
+  CU_ASSERT_EQUAL_FATAL(r, DDS_RETCODE_BAD_PARAMETER);
+  
+  printf("\n%s", buffer);
+}
+
+CU_Test(ddsrt_library, dlsym)
+{
+  dds_retcode_t r;
+  ddsrt_lib l;
+  void* f;
+
+  r = ddsrt_dlopen(TEST_LIB_NAME, true, &l);
+  CU_ASSERT_PTR_NOT_NULL(l);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_OK);
+  TEST_ABORT_IF_NULL(l, "ddsrt_dlopen() failed. Is the proper library path set?");
+
+  r = ddsrt_dlsym(l, "get_int", &f);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_OK);
+  CU_ASSERT_PTR_NOT_NULL(f);
+  TEST_ABORT_IF_NULL(f, "ddsrt_dlsym(l, \"get_int\") failed.");
+
+  r = ddsrt_dlclose(l);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_OK);
+}
+
+CU_Test(ddsrt_library, dlsym_bad)
+{
+  dds_retcode_t r;
+  ddsrt_lib l;
+  void* f;
+
+  r = ddsrt_dlopen(TEST_LIB_NAME, true, &l);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_OK);
+  CU_ASSERT_PTR_NOT_NULL(l);
+  TEST_ABORT_IF_NULL(l,"ddsrt_dlopen() failed. Is the proper library path set?");
+
+  r = ddsrt_dlsym(l, NULL, &f);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_BAD_PARAMETER);
+
+  r = ddsrt_dlsym(NULL, "UnknownSym", &f);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_BAD_PARAMETER);
+
+  r = ddsrt_dlclose(l);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_OK);
+}
+
+CU_Test(ddsrt_library, dlsym_unknown)
+{
+  char buffer[256];
+  dds_retcode_t r;
+  ddsrt_lib l;
+  void* f;
+
+  r = ddsrt_dlopen(TEST_LIB_NAME, true, &l);
+  CU_ASSERT_PTR_NOT_NULL(l);
+  TEST_ABORT_IF_NULL(l,"ddsrt_dlopen() failed. Is the proper library path set?");
+
+  r = ddsrt_dlsym(l, "UnknownSym", &f);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_ERROR);
+  CU_ASSERT_PTR_NULL_FATAL(f);
+
+  r = ddsrt_dlerror(buffer, sizeof(buffer));
+  CU_ASSERT_EQUAL_FATAL(r, DDS_RETCODE_OK);
+  printf("\n%s", buffer);
+
+  r = ddsrt_dlclose(l);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_OK);
+}
+
+typedef void (*func_set_int)(int val);
+typedef int  (*func_get_int)(void);
+CU_Test(ddsrt_library, call)
+{
+  int get_int = 0;
+  int set_int = 1234;
+  func_get_int f_get;
+  func_set_int f_set;
+  dds_retcode_t r;
+  ddsrt_lib l;
+
+  r = ddsrt_dlopen(TEST_LIB_NAME, true, &l);
+  CU_ASSERT_PTR_NOT_NULL(l);
+  TEST_ABORT_IF_NULL(l, "ddsrt_dlopen() failed. Is the proper library path set?");
+
+  r = ddsrt_dlsym(l, "get_int", (void **)&f_get);
+  CU_ASSERT_PTR_NOT_NULL(f_get);
+  TEST_ABORT_IF_NULL(f_get, "ddsrt_dlsym(l, \"get_int\") failed.");
+
+  r = ddsrt_dlsym(l, "set_int", (void **)&f_set);
+  CU_ASSERT_PTR_NOT_NULL(f_set);
+  TEST_ABORT_IF_NULL(f_set, "ddsrt_dlsym(l, \"set_int\") failed.");
+
+  f_set(set_int);
+  get_int = f_get();
+  CU_ASSERT_EQUAL(set_int, get_int);
+
+  r = ddsrt_dlclose(l);
+  CU_ASSERT_EQUAL(r, DDS_RETCODE_OK);
+}
+
+CU_Test(ddsrt_library, dlclose_null)
```

Comment:
Why is this called dlclose_null? There's only one test with NULL. Please split this up in multiple tests.